torchci/clickhouse_queries/ttrs_percentiles/query.sql

```diff
@@ -1,66 +1,53 @@
--- !!! Query is not converted to CH syntax yet.  Delete this line when it gets converted
 -- This query is used to compute the TTRS KPI for the pytorch/pytorch repo.
 --
 -- Results are displayed on HUD in two views:
 --   The kpi view, where percentile_to_get should be left at zero in order to get the default percentiles
 --   The metrics view, where the percentile_to_get and one_bucket should be set in order to get just the desired percentile
 --
 -- This query has two special params:
---     percentile_to_get: When set, it returns only the specified percentile. Otherwise it returns
---                 p25, p50, p75 and p90 percentiles.
+--     percentile_to_get: Custom percentile to get
 --     one_bucket: When set to false, buckets data into weekly percentiles. When true, it treats
 --                 entire time range AS one big bucket and returns percnetiles accordingly
 
 WITH
--- All the percentiles that we want the query to determine
-percentiles_desired AS (
-  SELECT
-    CONCAT('p', n.percentile) as percentile,
-    n.percentile / 100.0 as percentile_num
-  FROM  UNNEST(ARRAY_CREATE(25, 50, 75, 90) AS percentile) AS n
-  UNION ALL
-    -- if percentile_to_get is specified, we get and only return that percentile
-  SELECT
-    CONCAT(
-      'p',
-      CAST(
-        ROUND(: percentile_to_get * 100) AS STRING
-      )
-    ),
-    : percentile_to_get
-  WHERE
-    : percentile_to_get > 0
-),
 -- Get all PRs that were merged into master, and get all the SHAs for commits from that PR which CI jobs ran against
 -- We need the shas because some jobs (like trunk) don't have a PR they explicitly ran against, but they _were_ run against
 -- a commit from a PR
 pr_shas AS (
   SELECT
-    r.pull_requests[1].number AS pr_number,
+    r.pull_requests[1].'number' AS pr_number,
     CONCAT(
       'https://github.com/pytorch/pytorch/pull/',
-      r.pull_requests[1].number
+      r.pull_requests[1].'number'
     ) AS url,
-    j.head_sha AS sha,
+    j.head_sha AS sha
   FROM
-    commons.workflow_job j
-    INNER JOIN commons.workflow_run r ON j.run_id = r.id
+    default.workflow_job j final
+    INNER JOIN default.workflow_run r final ON j.run_id = r.id
   WHERE
     1 = 1
-    AND j._event_time > PARSE_DATETIME_ISO8601(: startTime)
-    AND r._event_time > PARSE_DATETIME_ISO8601(: startTime)
-    AND j._event_time < PARSE_DATETIME_ISO8601(: stopTime)
-    AND r._event_time < PARSE_DATETIME_ISO8601(: stopTime)
+    and j.id in (
+      select id from
+      materialized_views.workflow_job_by_started_at
+      where started_at > {startTime: DateTime64(3)}
+      and started_at < {stopTime: DateTime64(3)}
+    )
+    and r.id in (
+      select id from
+      materialized_views.workflow_run_by_run_started_at
+      where run_started_at > {startTime: DateTime64(3)}
+      and run_started_at < {stopTime: DateTime64(3)}
+    )
```

Comment:
Use materialized views as indexes to speed up query

Reply:
That's even faster than it was on Rockset IIRC